core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +377,302 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
+          new AuthorizationPluginCallbackWrapper();
+
+      RoleEntity updatedRole =
+          store.update(
+              AuthorizationUtils.ofRole(metalake, role),
+              RoleEntity.class,
+              Entity.EntityType.ROLE,
+              roleEntity -> {
+                List<SecurableObject> revokedSecurableObjects =
+                    generateNewSecurableObjects(
+                        roleEntity.securableObjects(),
+                        object,
+                        targetObject -> {
+                          // If the securable object doesn't exist, we do nothing except for
+                          // logging.
+                          if (targetObject == null) {
+                            LOG.warn(
+                                "Securable object {} type {} doesn't exist in the role {}",
+                                object.fullName(),
+                                object.type(),
+                                role);
+                            return null;
+                          } else {
+                            // If the securable object exists, we remove the privileges of the
+                            // securable object and will remove duplicated privileges at the same
+                            // time.
+                            return updateRevokedSecurableObject(
+                                metalake,
+                                role,
+                                object,
+                                privileges,
+                                roleEntity,
+                                targetObject,
+                                authorizationCallbackWrapper);
+                          }
+                        });
+
+                AuditInfo auditInfo =
+                    AuditInfo.builder()
+                        .withCreator(roleEntity.auditInfo().creator())
+                        .withCreateTime(roleEntity.auditInfo().createTime())
+                        .withLastModifier(PrincipalUtils.getCurrentPrincipal().getName())
+                        .withLastModifiedTime(Instant.now())
+                        .build();
+
+                return RoleEntity.builder()
+                    .withId(roleEntity.id())
+                    .withName(roleEntity.name())
+                    .withNamespace(roleEntity.namespace())
+                    .withProperties(roleEntity.properties())
+                    .withAuditInfo(auditInfo)
+                    .withSecurableObjects(revokedSecurableObjects)
+                    .build();
+              });
+
+      authorizationCallbackWrapper.execute();
+
+      return updatedRole;
+    } catch (NoSuchEntityException nse) {
+      LOG.error("Failed to revoke, role {} does not exist in the metalake {}", role, metalake, nse);
+      throw new NoSuchRoleException(ROLE_DOES_NOT_EXIST_MSG, role, metalake);
+    } catch (IOException ioe) {
+      throw new RuntimeException(ioe);
```

Comment:
I noticed that you have logged `NoSuchEntityException`. Why don't you also add logs here? Do you have any other considerations?

Reply:
Added the log.